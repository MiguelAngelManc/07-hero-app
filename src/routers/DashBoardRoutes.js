import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom/cjs/react-router-dom.min'
import { DcScreen } from '../Components/dc/DcScreen'
import { HeroScreen } from '../Components/heroes/HeroScreen'
import { MarvelScreen } from '../Components/marvel/MarvelScreen'
import { SearchScreen } from '../Components/search/SearchScreen'
import { Navbar } from '../Components/ui/Navbar'

export const DashBoardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className="container">

                <Switch>

                    <Route exact path="/marvel" component={ MarvelScreen }/>
                    <Route exact path="/hero/:heroeId" component={ HeroScreen }/>
                    <Route exact path="/dc" component={ DcScreen }/>
                    <Route path="/search" component={ SearchScreen }/>

                    <Redirect to="/marvel"/>
                    
                </Switch>

            </div>
        </>
    )
}
