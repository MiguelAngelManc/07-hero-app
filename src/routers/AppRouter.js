import React from "react";
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link
} from "react-router-dom";
import { DcScreen } from "../Components/dc/DcScreen";
import { LoginScreen } from "../Components/login/LoginScreen";
import { MarvelScreen } from "../Components/marvel/MarvelScreen";
import { Navbar } from "../Components/ui/Navbar";
import { DashBoardRoutes } from "./DashBoardRoutes";

export const AppRouter = () => {
    return (
        <Router>
            <div>

                <Switch>
                    <Route exact path="/login" component={ LoginScreen } />
                    
                    <Route path="/" component={ DashBoardRoutes } />
                    

                </Switch>
            </div>
        </Router>
    );
}
    
