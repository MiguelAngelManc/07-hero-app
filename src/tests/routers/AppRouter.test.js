import { mount } from "enzyme"
import { AuthContext } from "../../auth/AuthContext"
import { AppRouter } from "../../routers/AppRouter"

describe('Pruebas de AppRouter', () => {

    const contextValue = {
        dispatch:jest.fn(),
        user:{
            logged:false
        }
    }

    
    test('Mostrar login cuando no autentificado', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        )

        expect(wrapper).toMatchSnapshot();
        
    })
    
    test('Debe mostrar marvel si esta autentificado', () => {
        const wrapper = mount(
            <AuthContext.Provider value={{user:{logged:true,name:"miguel"}}}>
                <AppRouter />
            </AuthContext.Provider>
        )

        expect(wrapper.find('.navbar').exists()).toBe(true)

    })
    


})
