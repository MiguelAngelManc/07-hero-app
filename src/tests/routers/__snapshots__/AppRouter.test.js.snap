// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de AppRouter Mostrar login cuando no autentificado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <input
      autoComplete="off"
      name="username"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value=""
    />
    <button
      className="btn btn-primary btn-box"
      onClick={[Function]}
    >
      Login!
    </button>
  </div>
</div>
`;
